vnext/Shared/BaseScriptStoreImpl.cpp

```diff
@@ -297,6 +317,15 @@ void BasePreparedScriptStoreImpl::persistPreparedScript(
   prefix->runtimeVersion = runtimeMetadata.version;
   prefix->sizeInBytes = preparedScript->size();
 
+  std::optional<std::vector<std::uint8_t>> hashBuffer =
+      Microsoft::ReactNative::GetSHA256Hash(preparedScript->data(), preparedScript->size());
+  if (!hashBuffer) {
+    // Hashing failed.
+    std::terminate();
```

Comment:
Shouldn't we crash here so we get info for any investigation? From my understanding std::terminate doesn't produce Watson crash telemetry